test/distributed/test_c10d_common.py

```diff
@@ -1468,10 +1468,13 @@ def _call_collective_with_varying_tensors(self, backend, collective, *args):
             collective()
         elif collective == dist.all_gather:
             collective([tensor], tensor, *args)
-        elif collective == dist.reduce_scatter:
+        elif collective == dist.reduce_scatter or collective == dist.all_to_all:
+            # gloo does not support reduce_scatter or all_to_all
             if backend != "gloo":
-                # gloo does not support reduce_scatter
-                collective(tensor, [tensor], *args)
+                if collective == dist.reduce_scatter:
+                    collective(tensor, [tensor], *args)
+                else:
+                    collective([tensor], [tensor], *args)
```

Comment:
nit: please see my comment in [10/N] to dissolve this if-else block.

Reply:
Thanks! Agree with this, will update in future PR to avoid rebase